vsphere/host_network_policy_structure.go

```diff
@@ -0,0 +1,286 @@
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	hostNetworkPolicyNicTeamingPolicyModeLoadbalanceIP     = "loadbalance_ip"
+	hostNetworkPolicyNicTeamingPolicyModeLoadbalanceSrcMac = "loadbalance_srcmac"
+	hostNetworkPolicyNicTeamingPolicyModeLoadbalanceSrcID  = "loadbalance_srcid"
+	hostNetworkPolicyNicTeamingPolicyModeFailoverExplicit  = "failover_explicit"
+)
+
+var hostNetworkPolicyNicTeamingPolicyAllowedValues = []string{
+	hostNetworkPolicyNicTeamingPolicyModeLoadbalanceIP,
+	hostNetworkPolicyNicTeamingPolicyModeLoadbalanceSrcMac,
+	hostNetworkPolicyNicTeamingPolicyModeLoadbalanceSrcID,
+	hostNetworkPolicyNicTeamingPolicyModeFailoverExplicit,
+}
+
+// schemaHostNetworkPolicy returns schema items for resources that need to work
+// with a HostNetworkPolicy, such as virtual switches and port groups.
+func schemaHostNetworkPolicy() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		// HostNicTeamingPolicy/HostNicFailureCriteria
+		"check_beacon": &schema.Schema{
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Enable beacon probing. Requires that the vSwitch has been configured to use a beacon. If disabled, link status is used only.",
+		},
+
+		// HostNicTeamingPolicy/HostNicOrderPolicy
+		"active_nics": &schema.Schema{
+			Type:        schema.TypeList,
+			Description: "List of active network adapters used for load balancing.",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		},
+		"standby_nics": &schema.Schema{
+			Type:        schema.TypeList,
+			Description: "List of standby network adapters used for failover.",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		},
+
+		// HostNicTeamingPolicy
+		"teaming_policy": &schema.Schema{
+			Type:         schema.TypeString,
+			Optional:     true,
+			Description:  "The network adapter teaming policy. Can be one of loadbalance_ip, loadbalance_srcmac, loadbalance_srcid, or failover_explicit.",
+			ValidateFunc: validation.StringInSlice(hostNetworkPolicyNicTeamingPolicyAllowedValues, false),
+		},
+		"notify_switches": &schema.Schema{
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "If true, the teaming policy will notify the broadcast network of a NIC failover, triggering cache updates.",
+		},
+		"failback": &schema.Schema{
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "If true, the teaming policy will re-activate failed interfaces higher in precedence when they come back up.",
+		},
+
+		// HostNetworkSecurityPolicy
+		"allow_promiscuous": &schema.Schema{
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Enable promiscuous mode on the network. This flag indicates whether or not all traffic is seen on a given port.",
+		},
+		"allow_forged_transmits": &schema.Schema{
+			Type:        schema.TypeBool,
+			Optional:    true,
```

Comment:
Hey @tombuildsstuff - these are defaulted in `resource_host_virtual_switch.go`. They need to be non-defaulted here so that the schema can be shared with `vsphere_host_port_group` for inheritance purposes.